clang/lib/Sema/SemaChecking.cpp

```diff
@@ -2976,6 +2976,12 @@ Sema::CheckBuiltinFunctionCall(FunctionDecl *FDecl, unsigned BuiltinID,
     break;
 
   case Builtin::BI__builtin_ct_select: {
+    // check to see if the Arch supports it
+    if (!Context.getTargetInfo().hasFeature("ctselect")) {
```

Comment:
It looks good to me at the moment, but we do intent to provide fallback implementation for targets that does not support `ctselect`. This is one of thing you can look into next.